node/client/src/lib.rs

```diff
@@ -340,17 +361,35 @@ impl Client {
                 .remove(&next_beacon_block.body.header.shard_block_hash)
                 .expect(BEACON_SHARD_BLOCK_MATCH);
 
+            let bls_keys = self.get_authority_keys(next_beacon_block.index());
+            if !next_beacon_block.signature.verify(&bls_keys, next_beacon_block.hash.as_ref()) ||
+                !next_shard_block.signature.verify(&bls_keys, next_shard_block.hash.as_ref()) {
+                error!(target: "client", "Importing a block with an incorrect signature ({:?}, {:?}); signers: ({:?},{:?})",
+                       next_shard_block.block_hash(), next_beacon_block.block_hash(),
+                       next_beacon_block.signature.authority_count(),
+                       next_shard_block.signature.authority_count());
+                // TODO enable when we sign blocks with second BLS
+                if false {
```

Comment:
I do not think that's where we should be verifying the signatures of the block.
Verification should be happening before block is added to `pending_*_blocks`.